app/src/main/java/com/webkeyz/batchtwotraining/MainActivity.java

```diff
@@ -1,14 +1,48 @@
 package com.webkeyz.batchtwotraining;
 
+import android.os.Bundle;
+import android.util.Log;
+
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProviders;
+import androidx.paging.PagedList;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 
-import android.os.Bundle;
+import com.webkeyz.batchtwotraining.adapters.FeedAdapter;
+import com.webkeyz.batchtwotraining.models.ArticlesItem;
+import com.webkeyz.batchtwotraining.models.Feed;
+import com.webkeyz.batchtwotraining.network.FeedApi;
+import com.webkeyz.batchtwotraining.network.ServiceGenerator;
+import com.webkeyz.batchtwotraining.viewmodels.FeedViewModel;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class MainActivity extends AppCompatActivity {
+    private static final String TAG = "MainActivity";
+    private FeedViewModel feedViewModel;
+    private FeedAdapter feedAdapter;
+    private RecyclerView recyclerView;
 
     @Override
+
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        recyclerView = findViewById(R.id.recyclerView);
+        feedViewModel = ViewModelProviders.of(this).get(FeedViewModel.class);
```

Comment:
ViewModelProviders is depreacted , now you are creating viewmodel via factory pattern
because old way prevent us from passing argument to view model , view model default factory is already create instance of view model without args , so current way we create custom factory to each view model to control passing args
https://medium.com/koderlabs/viewmodel-with-viewmodelprovider-factory-the-creator-of-viewmodel-8fabfec1aa4f